File: src/bp/MRF/Message.java
package bp.MRF;

import bp.algorithms.queues.IdentifiedClass;

import java.util.Arrays;

/**
 * Created by vaksenov on 23.07.2019.
 */
public class Message extends IdentifiedClass {
    public int i, j;
    Edge e;
    public double[] logMu;
    public Message reverse;

    public Message(int id, int i, int j, Edge e) {
        this.id = id;
        this.i = i;
        this.j = j;

        this.e = e;
        this.logMu = new double[e.potentials[0].length];
        Arrays.fill(logMu, Math.log(1. / logMu.length));
    }

    public double getPotential(int vi, int vj) {
        return e.getPotential(i, j, vi, vj);
    }
}

File: message.cpp
#include <algorithm>
#include <cmath>

#include "message.h"

//function called Message within a struct called Message within a namespace called Message :)
Message::Message::Message(uint64_t i, uint64_t j, Edge& e)
  : i(i), j(j), e(e), reverse(nullptr)
{
    std::fill(logMu.begin(), logMu.end(), std::log(1.0 / LENGTH));
}
